agent/containermetadata/parse_metadata.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// parseNetworkMetadata parses the docker.NetworkSettings struct and
+// packages the desired metadata for JSON marshaling
+// Since we accept incomplete metadata fields, we should not return
+// errors here and handle them at this stage.
+func parseNetworkMetadata(settings *docker.NetworkSettings) (NetworkMetadata, error) {
+	// If network settings are unavailable, users should be notified in case
+	// this was not intended
+	if settings == nil {
```

Comment:
I'm not sure about this. It appears that NetworkSettings will be non-nil regardless of network mode. In addition, the child field NetworkSettings.Networks will also be non-nil. It appears to provide no IP information, but it appears this should always be available except in case of an error. 

Examples of this for "host" mode
https://github.com/aws/amazon-ecs-agent/pull/848#discussion_r130170872
Example of this for "none" mode
```
"NetworkSettings": {
            "Bridge": "",
            "SandboxID": "ae5f577b2e4f67d10cb4d223f932fec4f62cf0fe621b9d77a8a8caa9e1ed5229",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/ae5f577b2e4f",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "none": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "f2553e8fc676e2337b2e7eeaa1f4ce6d05203b504472c1020e906ea637156fd2",
                    "EndpointID": "e44e16490cc30e04ac5e01b7970d004e353986e569b99fffeab5bc96df1e2372",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,

```